### PR TITLE
issue #760 add mising enums

### PR DESCRIPTION
**Pull request by Michal-Babins**
added missing enums for Annotation Info File, Structural Annotation Stats Json, Error Corrected Reads, Assembly Info File

### CODE REVIEW

**Review by aclum — COMMENTED** [review]
I don't any of the files at this point should have GOLD-APs in the pattern match.  
*assembly.contigs.fasta is the 'Assembly Contigs' file not the scaffolds file. There is no scaffolds file publicly available when pulling from JDP API, that will only exist for NMDC generated assemblies. If we aren't renaming headers we also need to include the *contig_names_mapping.tsv file so there is a mapping between the assembly results and the annotation results.

File: src/schema/nmdc.yaml [hunk 1]
```diff
@@ -1198,6 +1210,12 @@ enums:
           file_name_pattern:
             tag: file_name_pattern
             value: '[GOLD-AP]_ko.tsv'
```
[the remark this answers is not in the document]
Comment (Michal-Babins) [reply src/schema/nmdc.yaml:1212]: Yes, this is just an annotation known between the two. @turbomam, may be able to expand more on this.

**Comment by Michal-Babins** [comment]
@aclum Added mappings file and fixed pattern for contigs file.

**Review by aclum — COMMENTED** [review]
@Michal-Babins  Would you update the scaffolds match pattern. Right now it is confusing as those are both the same. We don't expect scaffold files for JGI data but if we had them the pattern they would follow is 'assembly.scaffolds.fasta'

**Review by aclum — COMMENTED** [review]
@Michal-Babins  would you also fix the typo @scanon mentioned in chat for the spelling of genemark. These rows should all be 'Genemark', not Genmark
      Genmark Annotation GFF:
        text: Genmark Annotation GFF
        description: GFF3 format file with Genmark

**Comment by Michal-Babins** [comment]
Done

**Comment by turbomam** [comment]
> @turbomam or @aclum any more issues we need to address before we merge?

Yes. I'm applying some syntactic fixes and simplifications now. Will have a report in a few minutes.

**Comment by turbomam** [comment]
Questions:
- Can you confirm that you want the software published at [this site](http://exon.gatech.edu/GeneMark/) to be labelled as Genemark, not GeneMark?
- Can you confirm that the filenames for `Crisprt Terms` files should contain the string 'crisprs', not 'crisprt'?

Steps I took
- Backmerge of main and local testing. PR becomes larger, but the only file we have to monitor is still just nmdc.yaml
- Fixed at least one indentation error, which was causing the PR check failure
- General improvements to spacing around and within each permissible value block 
- Global spellcheck
- Terser notation for permissible values, as suggested by @sujaypatil96 
  - no `text` line
  - compact, one-line-per-annotation format

Future work (I will make issues):
- what notation are you using for the `file_name_pattern`s? I get the feeling that '[GOLD-AP]_cath_funfam.gff' isn't meant to be taken either literally or as a regular expression. If '[GOLD-AP]' is meant to be symbolic for a variable portion of the file name, then we should reflect that with either named capture groups, or by renaming the annotation to something like `file_name_abstraction` or `file_name_formula`, making it clear that they are not regular expression patterns.
- the `description` annotations should add understanding beyond the permissible value name. There are several `description`s that almost exactly paraphrase the permissible value name. I'll make a report.

**Comment by Michal-Babins** [comment]
@turbomam I can confirm Crisprt --> Crisprs, and the conversation I had with Shane was Genmark --> Genemark, not sure about GeneMark.

**Comment by turbomam** [comment]
> @turbomam I can confirm Crisprt --> Crisprs, and the conversation I had with Shane was Genmark --> Genemark, not sure about GeneMark.

Ok. The publisher calls it GeneMark, but I will leave it as Genemark based on your team discussions. 

Sorry, that last PR of mine really was huge and makes it hard to track changes. If you can tell me how to remove the irrelevant stuff, I'm glad to do that.

**Comment by aclum** [comment]
GeneMark would be more proper, im not sure if we enforce or try to enforce proper case for all software tool names in the schema.

**Comment by turbomam** [comment]
closed in favor of 
- #778

**Pull request closed without merging**